### DISCUSSION TITLE
Cannot flash binary to ESP32Vroom

### DISCUSSION THREAD

**Discussion by rlust** [HASwitchPlate/openHASP#189] [Q&A] [answered]
## Questions
Which binary do I use for a ESP32 WROOM DEV module?
I am trying freetouchdeck_full_4MB_v0.6.0.bin

Chip Info:

Chip Family: ESP32
Chip Model: ESP32-D0WDQ6 (revision 1)
Number of Cores: 2
Max CPU Frequency: 240MHz
Has Bluetooth: YES
Has Embedded Flash: NO
Has Factory-Calibrated ADC: YES
MAC Address: 24:0A:C4:EC:26:CC
Uploading stub...
Running stub...
Stub running...
Changing baud rate to 460800
Changed.
Flash Size: 4MB
Unexpected error: The firmware binary is invalid (magic byte=FF, should be E9)
I am getting the above error.
We only keep bug reports and active requests being worked on in the issues tracker.

For general questions we have opened the [discussion forum](https://github.com/HASwitchPlate/openHASP/discussions) on github and [#openHASP channel](https://discord.gg/VCWyuhF) on Discord.

If you are not sure if your question is related to the code development, feel free to ask!

**Comment by fvanroie** [comment, marked as answer]
What display are you using? Typically you would use the firmware build depending on which dev module + display type.
So we also need to know the model or touchscreen you have connected.

The error you receive is typically seen in ESP Home flasher. Which program are you using the flash the ESP32?

**Reply by rlust** [reply]
I am using a ILI9488 display and a Dustin Watts ESP32 TFT Combiner. 
To flash I am using ESPHome Flasher.
What flasher do you suggest?

**Reply by fvanroie** [reply]
`freetouchdeck_full_4MB_v0.6.0.bin` is the correct binary for that combination.
I suggest using [Tasmota PyFlasher](https://haswitchplate.github.io/openHASP-docs/0.6/installation/esp32/#flash-esp32_1) for easy installation.

**Reply by rlust** [reply]
I was able to get it flashed using NodeMCU PyFlasher!! 

openHasp.log

**Comment by rlust** [comment]
[OpenHasp .log](https://github.com/HASwitchPlate/openHASP/files/6873932/OpenHasp.log)

**Reply by rlust** [reply]
Thanks for the help! This looks like a great project.
Here is my log file. I am getting an error "failed to open /littlefs/config.json"

**Reply by fvanroie** [reply]
The `config.json` file is saved after you configure the wifi `ssid` and `password` and reboot the plate.
So at first boot of an unconfigured plate, this is normal and can be ignored.

**Comment by fvanroie** [comment]
Thank you for using the "Mark as answer" button if the issue has been resolved.
